src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -354,51 +323,57 @@ uct_cuda_ipc_md_open(uct_component_t *component, const char *md_name,
         .mem_attach         = ucs_empty_function_return_unsupported,
         .detect_memory_type = ucs_empty_function_return_unsupported
     };
+    uct_md_t *md;
 
-    uct_md_t *md = ucs_calloc(1, sizeof(*md), "uct_cuda_ipc_md");
+    md = ucs_calloc(1, sizeof(*md), "uct_cuda_ipc_md");
     if (md == NULL) {
         return UCS_ERR_NO_MEMORY;
     }
 
     md->ops       = &md_ops;
-    md->component = &uct_cuda_ipc_component;
+    md->component = &uct_cuda_ipc_component.super;
     *md_p         = md;
     return UCS_OK;
 }
 
-uct_component_t uct_cuda_ipc_component = {
-    .query_md_resources = uct_cuda_base_query_md_resources,
-    .md_open            = uct_cuda_ipc_md_open,
-    .cm_open            = ucs_empty_function_return_unsupported,
-    .rkey_unpack        = uct_cuda_ipc_rkey_unpack,
-    .rkey_ptr           = ucs_empty_function_return_unsupported,
-    .rkey_release       = uct_cuda_ipc_rkey_release,
-    .rkey_compare       = uct_base_rkey_compare,
-    .name               = "cuda_ipc",
-    .md_config          = {
-        .name           = "Cuda-IPC memory domain",
-        .prefix         = "CUDA_IPC_",
-        .table          = uct_cuda_ipc_md_config_table,
-        .size           = sizeof(uct_cuda_ipc_md_config_t),
+uct_cuda_ipc_component_t uct_cuda_ipc_component = {
+    .super = {
+        .query_md_resources = uct_cuda_base_query_md_resources,
+        .md_open            = uct_cuda_ipc_md_open,
+        .cm_open            = ucs_empty_function_return_unsupported,
+        .rkey_unpack        = uct_cuda_ipc_rkey_unpack,
+        .rkey_ptr           = ucs_empty_function_return_unsupported,
+        .rkey_release       = uct_cuda_ipc_rkey_release,
+        .rkey_compare       = uct_base_rkey_compare,
+        .name               = "cuda_ipc",
+        .md_config          = {
+            .name           = "Cuda-IPC memory domain",
+            .prefix         = "CUDA_IPC_",
+            .table          = uct_cuda_ipc_md_config_table,
+            .size           = sizeof(uct_cuda_ipc_md_config_t),
+        },
+        .cm_config          = UCS_CONFIG_EMPTY_GLOBAL_LIST_ENTRY,
+        .tl_list            = UCT_COMPONENT_TL_LIST_INITIALIZER(&uct_cuda_ipc_component.super),
+        .flags              = 0,
+        .md_vfs_init        =
+                (uct_component_md_vfs_init_func_t)ucs_empty_function
     },
-    .cm_config          = UCS_CONFIG_EMPTY_GLOBAL_LIST_ENTRY,
-    .tl_list            = UCT_COMPONENT_TL_LIST_INITIALIZER(&uct_cuda_ipc_component),
-    .flags              = 0,
-    .md_vfs_init        = (uct_component_md_vfs_init_func_t)ucs_empty_function
+    .uuid_hash              = KHASH_STATIC_INITIALIZER,
+    .lock                   = {0}
```

Comment:
it's not correct on all systems, sometimes the unlocked value is -1
must switch to pthread_mutex_t

Reply:
ok